src/main/asciidoc/configuring-cors.adoc

```diff
@@ -0,0 +1,71 @@
+[[customizing-sdr.configuring-cors]]
+= Configuring CORS
+
+For security reasons, browsers prohibit AJAX calls to resources residing outside the current origin. When working with client-side HTTP requests issued by a browser you want to enable specific HTTP resources to be accessible.
+
+Spring Data REST supports as of 2.6 http://en.wikipedia.org/wiki/Cross-origin_resource_sharing[Cross-origin resource sharing] (CORS) through http://docs.spring.io/spring/docs/{version}/spring-framework-reference/html/cors.html[Spring's CORS] support.
+
+
+== Repository interface CORS configuration
+
+You can add a `@CrossOrigin` annotation to your repository interfaces to enable CORS for the whole repository. By default `@CrossOrigin` allows origins and `HEAD` and `GET` HTTP methods:
```

Comment:
As mentioned here, the current behaviour with Spring Data REST + `@CrossOrigin` is to allow `GET` and `HEAD` HTTP methods (default for the low level `CorsConfiguration` class).

In Spring MVC, the default behaviour for `@CrossOrigin` annotations without any method specified is to use methods of the related `@RequestMapping` annotation. I am wondering if by default in Spring Data REST, we could not just allow HTTP methods of the related REST endpoint, in order to avoid surprising the user when for example his repository methods exposed with a `POST` HTTP method will not work by default. Any thoughts?